ownCloud/Client/ClientRootViewController.swift

```diff
@@ -155,32 +166,28 @@ class ClientRootViewController: UITabBarController {
 
 		self.tabBar.applyThemeCollection(Theme.shared.activeCollection)
 
-		self.viewControllers = [filesNavigationController] as? [UIViewController]
-	}
-
-	func logoutBarButtonItem() -> UIBarButtonItem {
-		let barButton = UIBarButtonItem(title: "Disconnect".localized, style: .plain, target: self, action: #selector(logout(_:)))
-		barButton.accessibilityIdentifier = "disconnect-button"
-		return barButton
-	}
-
-	@objc func logout(_: Any) {
-		self.closeClient()
+        self.viewControllers = [filesNavigationController] as? [UIViewController]
 	}
 
 	func closeClient(completion: (() -> Void)? = nil) {
-		self.presentingViewController?.dismiss(animated: true, completion: completion)
+        self.navigationController?.setNavigationBarHidden(false, animated: false)
+        self.navigationController?.popViewController(animated: true, completion: {
+            
+        })
 	}
 
 	func coreReady() {
 		OnMainThread {
 			let queryViewController = ClientQueryViewController(core: self.core!, query: OCQuery(forPath: "/"))
-
-			queryViewController.navigationItem.leftBarButtonItem = self.logoutBarButtonItem()
-
-			self.filesNavigationController?.pushViewController(queryViewController, animated: false)
+            self.filesNavigationController?.setViewControllers([self.emptyViewController, queryViewController], animated: false)
 		}
 	}
+    
+    func navigationController(_: UINavigationController, willShow: UIViewController, animated: Bool) {
+        if willShow.isEqual(emptyViewController) {
```

Comment:
@hosy I would at least explain in a comment what emptyViewController is for to make it less confusing

Reply:
if the emptyViewController will show, because the push button in ClientQueryViewController was triggered, push immediately to the ServerListTableViewController, because emptyViewController is only a helper for showing the "Back" button in ClientQueryViewController

Reply:
@hosy I am fine with that approach, I just would like to have a comment in the code explaining why an empty view controller is used

Reply:
I added the comment above also to the code